### PR TITLE
Consolidate udev rules into a single file

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Merged the udev rules files into a single file, and tweaked it a bit to include all USB devices under the known Caterina VIDs, and add some missing rules (USBAspLoader, SAM-BA).

Also, while I was here, I removed some outdated FAQ items.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Erovia** [comment]
What's the advantage of having one file instead of having one file per bootloader?

**Comment by fauxpark** [comment]
Less copypasting, and it should be easier for users who don't know what their board's bootloader is.
I'm not sure what the advantage of having multiple files would be - the unused rules shouldn't interfere with non-bootloader devices anyway.

File: lib/python/qmk/cli/doctor.py [hunk 1]
```diff
@@ -182,18 +195,17 @@ def check_udev_rules():
...
 
         # Check if the desired rules are among the currently present rules
-        for bootloader, rules in desired_rules.items():
+        for rules in desired_rules.items():
             # For caterina, check if ModemManager is running
-            if bootloader == "caterina":
-                if check_modem_manager():
-                    ok = False
-                    cli.log.warn("{bg_yellow}Detected ModemManager without the necessary udev rules. Please either disable it or set the appropriate udev rules if you are using a Pro Micro.")
+            if check_modem_manager():
```
Comment (Erovia) [inline lib/python/qmk/cli/doctor.py:200]: I'd rather keep the old behaviour. It doesn't make much sense to run the MM check for every rule when it only makes sense for the Caterina ones.

File: lib/python/qmk/cli/doctor.py [hunk 2]
```diff
@@ -182,18 +195,17 @@ def check_udev_rules():
...
-                deprecated_rule = deprecated_rules.get(bootloader)
-                if deprecated_rule and deprecated_rule.issubset(current_rules):
-                    cli.log.warn("{bg_yellow}Found old, deprecated udev rules for '%s' boards. The new rules on https://docs.qmk.fm/#/faq_build?id=linux-udev-rules offer better security with the same functionality.", bootloader)
-                else:
-                    cli.log.warn("{bg_yellow}Missing udev rules for '%s' boards. You'll need to use `sudo` in order to flash them.", bootloader)
+                for bootloader, rule in deprecated_rules.items():
+                    if rule and rule.issubset(current_rules):
+                        cli.log.warn("{bg_yellow}Found old, deprecated udev rules for '%s' boards. The new rules on https://docs.qmk.fm/#/faq_build?id=linux-udev-rules offer better security with the same functionality.", bootloader)
+                    else:
+                        cli.log.warn("{bg_yellow}Missing udev rules for '%s' boards. You'll need to use `sudo` in order to flash them.", bootloader)
```
Comment (Erovia) [inline lib/python/qmk/cli/doctor.py:208]: I've been meaning to add the FAQ link to this warning as well.
What do you think?

File: lib/python/qmk/cli/doctor.py (hunk 1 above)
Comment (fauxpark) [reply lib/python/qmk/cli/doctor.py:200]: I'll be honest, I don't really know what I'm doing with this section. I have to actually test it. I was trying to keep as much of the existing functionality as possible while still accounting for the fact `desired_rules` no longer has a `bootloader` part.

Comment (fauxpark) [reply lib/python/qmk/cli/doctor.py:200]: Hm, doctor does not have the ability to write out `desired_rules` to a file, right? In that case keeping the bootloader dicts would make more sense.

**Review by Erovia — APPROVED** [review]

**Pull request merged**